File: lib/models/lesson.dart
class Lesson {
  final String id;
  final String title;
  final String courseId;

  Lesson({
    required this.id,
    required this.title,
    required this.courseId,
  });

  factory Lesson.fromJson(Map<String, dynamic> json) {
    return Lesson(
      id: json['_id'] ?? json['id'],
      title: json['title'],
      courseId: json['courseId'],
    );
  }
}

File: backend/src/models/Lesson.js
import mongoose from "mongoose";

const lessonSchema = new mongoose.Schema(
  {
    courseId: {
      type: String, // 🔥 string course id
      required: true,
    },

    title: {
      type: String,
      required: true,
    },

    content: {
      type: String,
      default: "",
    },

    // 🔥 FIX: string because auth uses "lect1"
    createdBy: {
      type: String,
      required: true,
    },
  },
  { timestamps: true }
);

// ✅ Prevent OverwriteModelError
export default mongoose.models.Lesson ||
  mongoose.model("Lesson", lessonSchema);
